packages/adapters-library/src/adapters/aave-v3/products/rewards/aaveV3RewardsAdapter.ts

```diff
@@ -0,0 +1,391 @@
+      this.provider,
+    )
+
+    /**
+     * Fake protocol token created to satisfy return type
+     */
+    this.INCENTIVES_CONTRACT_DETAILS = {
+      address: this.INCENTIVES_CONTRACT_ADDRESSES[this.chainId],
+      name: 'Aave V3 Rewards',
+      symbol: 'Rewards',
+      decimals: 18,
+    }
+  }
+
+  getProtocolDetails(): ProtocolDetails {
+    return {
+      protocolId: this.protocolId,
+      name: 'AaveV3',
+      description: 'AaveV3 defi adapter',
+      siteUrl: 'https:',
+      iconUrl: AAVE_ICON_URL,
+      positionType: PositionType.Supply,
+      chainId: this.chainId,
+      productId: this.productId,
+    }
+  }
+
+  /**
+   *
+   * AToken addresses required to query reward balances
+   */
+  async getAaveTokenAddresses(): Promise<string[]> {
+    const [aTokens, stableDebtTokens, variableDebtTokens] = await Promise.all([
+      this.helpers.metadataProvider.getMetadata({
+        protocolId: Protocol.AaveV3,
+        productId: 'a-token',
+      }),
+      this.helpers.metadataProvider.getMetadata({
+        protocolId: Protocol.AaveV3,
+        productId: 'stable-debt-token',
+      }),
+      this.helpers.metadataProvider.getMetadata({
+        protocolId: Protocol.AaveV3,
+        productId: 'variable-debt-token',
+      }),
+    ])
+    return [
+      ...aTokens.map((token) => token.address),
+      ...stableDebtTokens.map((token) => token.address),
+      ...variableDebtTokens.map((token) => token.address),
+    ]
+  }
+
+  @CacheToDb
+  async getProtocolTokens(): Promise<ProtocolToken[]> {
+    const rewardTokenAddresses = await this.incentivesContract.getRewardsList()
+
+    const rewardTokens = await Promise.all(
+      rewardTokenAddresses.map((rewardToken) => {
+        return this.helpers.getTokenMetadata(rewardToken)
+      }),
+    )
+
+    return [
+      { ...this.INCENTIVES_CONTRACT_DETAILS, underlyingTokens: rewardTokens },
+    ]
+  }
+
+  private detectPositionEventSignature(
+    eventSignature = 'Transfer(address,address,uint256)',
+  ): string {
+    return ethers.id(eventSignature)
+  }
+
+  /**
+   * Checks if the user has ever opened a position in AaveV3
+   * Return AToken addresses if found
+   */
+  private async openPositions(userAddress: string): Promise<string[]> {
+    const topic0 = this.detectPositionEventSignature()
+    const topic1 = ethers.zeroPadValue(ZERO_ADDRESS, 32)
+    const topic2 = ethers.zeroPadValue(userAddress, 32)
+
+    const logs = await this.provider.getLogs({
+      topics: [topic0, topic1, topic2],
+      fromBlock: '0x',
+      toBlock: 'latest',
+    })
+
+    const aTokenAddresses = await this.getAaveTokenAddresses()
+
+    const filteredLogs = logs
+      .filter((log) => aTokenAddresses.includes(log.address))
+      .map((log) => log.address)
+
+    return [...new Set(filteredLogs)]
+  }
+
+  async getPositions(input: GetPositionsInput): Promise<ProtocolPosition[]> {
+    const { userAddress, blockNumber, protocolTokenAddresses } = input
```

Comment:
I don't think we are using the `protocolTokenAddresses` filter.

Reply:
ah i should try to use it i think e.g. return if not this protocolTokenAddress